File: python/basic_algorithm/former/quickSort.py
def quickSort(num,lower,upper):
    if lower >= upper:
        return num
    provit = num[lower]
    left = lower+1
    right = upper
    while left<=right:
        while left<=right and num[left]< provit:
            left+=1
        while left<=right and num[right]>=provit:
            right-=1
        if left>right:
            break
        num[left],num[right] = num[right],num[left]
    num[lower],num[right] = num[right],num[lower]
    quickSort(num,lower,right-1)
    quickSort(num,right+1,upper)
    return num


a = [2,3,1,5,6,3,2,8,6,0]
quickSort(a,0,len(a)-1)
print(a)

File: java/basic_algorithm/former/quick_sort.java
public class quick_sort{

    public static void quick_sort(int[] intList,int low,int up){
        if(low>up)return;
        int privot = low;
        int left = low+1;
        int right = up;
        while(left<=right){
            while(left<=right&&intList[left]<intList[privot]){
                left++;
            }
            while(left<=right&&intList[right]>=intList[privot]){
                right--;
            }
            if(left>right)break;
            int tmp = intList[left];
            intList[left] = intList[right];
            intList[right] = tmp;
        }
        int tmp = intList[right];
        intList[right] = intList[privot];
        intList[privot] = tmp;
        quick_sort(intList,low,privot-1);
        quick_sort(intList,privot+1,up);


    }
    public static void main(String[] args){
        int[] intList = new int[]{4,5,6,7,8,1,2,5};
        quick_sort(intList,0,7);
    }
}